src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -148,27 +191,67 @@ public static string HtmlDecode(string value)
                 return value;
             }
 
-            // Don't create StringBuilder if we don't have anything to encode
             if (!StringRequiresHtmlDecoding(value))
             {
                 return value;
             }
 
-            StringBuilder sb = StringBuilderCache.Acquire(value.Length);
-            HtmlDecode(value, sb);
-            return StringBuilderCache.GetStringAndRelease(sb);
+            char[] rawBuffer = null;
+            // We expect the encoded pattern to be at least twice as large as the decoded one.
```

Comment:
I think that this is wrong assumption. I think that there are going to be just a very few escaped characters in the typical case.

Reply:
okay, I will trim the check down to 256. do you think the 80 for the encode check is conservative enough or should we maybe even increase it to 100 / 128?